File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numberSet = new HashSet<>();
        
        for (int num : nums) {
            numberSet.add(num);
        }
        int longest = 0;
        for (int num : nums) {
            if (!numberSet.contains(num - 1)) {
                int length = 1;
                while (numberSet.contains(num + length)) {
                    length++;
                }
                longest = Math.max(longest, length);
            }
        }
        return longest;
               
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-3.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        currentEnd = farthest = 0
        jump = 0
        n = len(nums)

        for i in range(n-1):
            farthest = max(farthest, i + nums[i])
            if i == currentEnd:
                jump += 1
                currentEnd = farthest
            if currentEnd >= n-1:
                break
        return jump            